File: tests/transpiler/x/java/group_by_having.java
public class Main {
    static Data1[] people = new Data1[]{new Data1("Alice", "Paris"), new Data1("Bob", "Hanoi"), new Data1("Charlie", "Paris"), new Data1("Diana", "Hanoi"), new Data1("Eve", "Paris"), new Data1("Frank", "Hanoi"), new Data1("George", "Paris")};
    static class Data1 {
        String name;
        String city;
        Data1(String name, String city) {
            this.name = name;
            this.city = city;
        }
    }

    static java.util.List<Result4> big = new java.util.ArrayList<Result4>() {{ java.util.LinkedHashMap<String,Group2> _groups = new java.util.LinkedHashMap<>(); for (var p : people) { var _k = p.city; String _ks = String.valueOf(_k); Group2 g = _groups.get(_ks); if (g == null) { g = new Group2(_k, new java.util.ArrayList<>()); _groups.put(_ks, g); } g.items.add(p); } for (var g : _groups.values()) { if (g.items.size() >= 4) { add(new Result4(g.key, g.items.size())); } }}};
    static class Group2 {
        String key;
        java.util.List<Data1> items;
        Group2(String key, java.util.List<Data1> items) {
            this.key = key;
            this.items = items;
        }
    }

    static class Result4 {
        Object city;
        int num;
        Result4(Object city, int num) {
            this.city = city;
            this.num = num;
        }
    }


    public static void main(String[] args) {
        json(big);
    }
}

File: tests/transpiler/x/kt/group_by_having.kt
data class GGroup(val key: Any, val items: MutableList<MutableMap<String, String>>)
fun main() {
    val people = mutableListOf(mutableMapOf("name" to "Alice", "city" to "Paris"), mutableMapOf("name" to "Bob", "city" to "Hanoi"), mutableMapOf("name" to "Charlie", "city" to "Paris"), mutableMapOf("name" to "Diana", "city" to "Hanoi"), mutableMapOf("name" to "Eve", "city" to "Paris"), mutableMapOf("name" to "Frank", "city" to "Hanoi"), mutableMapOf("name" to "George", "city" to "Paris"))
    val big = run {
    val _groups = mutableMapOf<Any, MutableList<MutableMap<String, String>>>()
    for (p in people) {
        val _list = _groups.getOrPut(p["city"] as Any) { mutableListOf<MutableMap<String, String>>() }
        _list.add(p)
    }
    val _res = mutableListOf<MutableMap<String, Any>>()
    for ((key, items) in _groups) {
        val g = GGroup(key, items)
        if (g.items.size >= 4) {
            _res.add(mutableMapOf("city" to g.key, "num" to g.items.size))
        }
    }
    _res
}
    json(big)
}
